lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -896,8 +898,10 @@ getWalletWithCreationTime ctx (ApiT wid) = do
         , state =
             ApiT $ meta ^. #status
         , tip = ApiBlockReference
-            { epochNumber = (currentTip wallet) ^. #slotId . #epochNumber
-            , slotNumber =  (currentTip wallet) ^. #slotId . #slotNumber
+            { epochNumber =
+                unEpochNo $ (currentTip wallet) ^. #slotId . #epochNumber
+            , slotNumber =
+                unSlotNo $ (currentTip wallet) ^. #slotId . #slotNumber
```

Comment:
If we want to derive JSON instances for `ApiBlockReference` via `Generic`, we need `ToJSON` and `FromJSON` instances for `EpochNo` and `SlotNo`, which we cannot have.

We could make it `ApiT EpochNo` and `ApiT SlotNo`, but then, I think, the raw `Word16`, `Word64` seems better to me.

Reply:
In that case, I suspect it would be better to write:
```hs
data ApiBlockReference = ApiBlockReference
    { epochNumber :: !(ApiT EpochNo)
    , slotNumber :: !(ApiT SlotNo)
    , height :: !(Quantity "block" Natural)   
    } deriving (Eq, Generic, Show)
```
The advantage of doing this is that we can get better encapsulation and avoid repetition.

In particular, if we want to change the definition of `EpochNo` from `Word16` to `Word32` at a later point in time, we will only need to do it in one place.